app/scripts/modules/amazon/src/pipeline/stages/deployCloudFormation/evaluateCloudFormationChangeSetExecutionApproval.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+
+import { IExecution, IExecutionStage } from 'core/domain';
+import { Application } from 'core/application/application.model';
+import { NgReact } from '@spinnaker/core';
+import { AwsReactInjector } from 'amazon/reactShims';
+
+export interface IEvaluateCloudFormationChangeSetExecutionApprovalProps {
+  execution: IExecution;
+  stage: IExecutionStage;
+  application: Application;
+}
+
+export interface IEvaluateCloudFormationChangeSetExecutionApprovalState {
+  submitting: boolean;
+  judgmentDecision: string;
+  error: boolean;
+}
+
+export class EvaluateCloudFormationChangeSetExecutionApproval extends React.Component<
```

Comment:
I recommend a functional component utilizing hooks. You can look at any file that has `useState` for an example of this. Some recent files are the `SlackChannelSelector` and the `ManagedResourceConfig`.

Reply:
👍 changed to functional component with hooks